flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionFactory.java

```diff
@@ -215,10 +246,19 @@ private static void releasePartitionsQuietly(ResultSubpartition[] partitions, in
 		return () -> {
 			int maxNumberOfMemorySegments = type.isBounded() ?
 				numberOfSubpartitions * networkBuffersPerChannel + floatingNetworkBuffersPerGate : Integer.MAX_VALUE;
+			int numRequiredBuffers = numberOfSubpartitions + 1;
+
+			if (isSortMergeBlockingShuffle(type, numberOfSubpartitions)) {
```

Comment:
Maybe rewrite this to not compute it twice in the sort shuffle mode (once for non-sort shuffle, then for sort shuffle).